test/scapy/layers/dns.uts

```diff
@@ -422,3 +422,9 @@ assert p == eval(p.command())
 
 p = DNS(qd=[])
 assert p == eval(p.command())
+
+= DNS - iter through DNSStrFields
+
+pkt = DNSQR(qname=["domain1.com", "domain2.com"], qtype="A")
+for i in pkt:
+    assert i.qname in [b"domain1.com.", b"domain2.com."]
```

Comment:
Thanks!

As far as I understand it partly fixes the issue because `DNSQR` don't seem to be unrolled when they are inside `DNS` packets.
```python
>>> p = DNS(qd=DNSQR(qname=[b"domain1.com", b"domain2.com"], qtype="A"))
>>> sendp(IP()/UDP()/p)
.
Sent 1 packets.
```

With `scapy-2.4.3` it sends 2 packets:
```python
>>> p = DNS(qd=DNSQR(qname=[b"domain1.com", b"domain2.com"], qtype="A"))
>>> sendp(IP()/UDP()/p)
..
Sent 2 packets.
```

(it could be I misunderstood what that issue is about though. Sorry!)

Reply:
Fixed, thanks.

Reply:
For the record:
```python
p = DNS(qd=DNSQR(qname=[b"domain1.com", b"domain2.com"], qtype="A"))
```
will now be aliased to the following to keep <2.4.3 compatibility:
```python
p = DNS(qd=[DNSQR(qname=[b"domain1.com"], qtype="A"), DNSQR(qname=[b"domain2.com"], qtype="A")])
```

This does not mean that the following works:
```python
p = DNS(qd=[DNSQR(qname=[b"domain1.com", b"domain2.com"], qtype="A")])
```
Indeed it would create too much nesting if we did expand packets from within a list.

So in 2.5.0+ format, and more generally when using lists (even in 2.4.3), this is what should be used:
```python
p = DNS(qd=(DNSQR(qname=x, qtype="A") for x in [b"domain1.com", b"domain2.com"]))
```

Reply:
Hm this is not trivial.

Reply:
FWIW having thought about this now that `qd` is a list of `DNSQR`s
```python
DNSQR(qname=[b"domain1.com", b"domain2.com"], qtype="A")
```
looks kind of ambiguous to me. It could legitimately expand into two separate packets or into one packet with two different queries (which would be great with mDNS). I have to admit I'm not sure how it should work. Maybe qnames should be just strings after all. All the ambiguities can be sorted out explicitly. I don't know.

Reply:
I agree it shouldn't block this PR (and I'm kind of get used to flattening those lists myself).

Whether DNS packets should be unrolled or not is probably a separate question too (it's just that I looked at the "would then be iterated over when sending packets" part of that issue and wasn't sure what should happen).

I think what's great is that with this PR applied it's now possible to use something like
```python
DNSRR(rrname=['H.local', 'C.local'], ttl=[2, 0])
```
to produce 4 RRs to inject into one packet.